### PR TITLE
Add multipart configuration options to servlet extension

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

File: extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java [hunk 1]
```diff
@@ -87,6 +92,37 @@ public String getEffectiveServletName(final String key) {
...
+             * size.
+             */
+            @ConfigItem(defaultValue = "-1")
+            public Optional<Long> maxRequestSize;
+
+            /**
+             * The file size in bytes after which the file will be temporarily stored on disk.
+             */
+            @ConfigItem(defaultValue = "0")
+            public Optional<Integer> fileSizeThreshold;
```
Comment (ppalaga) [inline extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java:124]: I wonder whether all these need be optional given that they all have defaults? The optionals will always be non-empty. 

I also wonder why fileSizeThreshold is int while the others are long?

File: integration-tests/servlet/src/main/java/org/apache/camel/quarkus/component/servlet/MultiPartProcessor.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@Singleton
+public class MultiPartProcessor implements Processor {
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        // TODO: Ideally AttachmentMessage would be used here instead of plain Servlet APIs
+        // https://issues.apache.org/jira/browse/CAMEL-19899
+        Message message = exchange.getMessage();
+        HttpServletRequest request = message.getHeader(Exchange.HTTP_SERVLET_REQUEST, HttpServletRequest.class);
+        InputStream file = request.getPart("file").getInputStream();
+        message.setBody(file.readAllBytes(), String.class);
```
Comment (ppalaga) [inline integration-tests/servlet/src/main/java/org/apache/camel/quarkus/component/servlet/MultiPartProcessor.java:36]:
```suggestion
        try (InputStream file = request.getPart("file").getInputStream()) {
            message.setBody(file.readAllBytes(), String.class);
        }
```

**Review by oscerd — APPROVED** [review]

File: extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java (hunk 1 above)
Comment (jamesnetherton) [reply extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java:124]: >  The optionals will always be non-empty

I can remove the `Optional`.

> I also wonder why fileSizeThreshold is int while the others are long

See [MultipartConfigElement](https://jakarta.ee/specifications/servlet/5.0/apidocs/jakarta/servlet/multipartconfigelement).

**Comment by jamesnetherton** [comment]
Latest commit should address review feedback.

**Comment by jamesnetherton** [comment]
Setting the PR to draft as there's an odd issue happening on GitHub CI that I cannot replicate locally. Needs more investigation.

**Comment by jamesnetherton** [comment]
Issues are hopefully resolved now. Turns out I cannot count properly. I had set the byte limit in the config props to less than what I was sending in the valid test scenarios.

**Review by aldettinger — APPROVED** [review]

**Comment by jamesnetherton** [comment]
To move this along I opened https://github.com/apache/camel-quarkus/issues/5830 to follow up on later.

**Pull request merged**